### PR TITLE
Sticky top banner

### PR DESCRIPTION
**Pull request by regiskuckaertz**
Enables native sticky positioning for the top banner.

**I removed an `overflow-x` rule on the `html` and `body` elements** that was allegedly there because of annoying ads—I failed to find an example though.

TODO:
- [x] test in browsers with no support for sticky
- [x] merge #15510 
- [x] add animated GIFs :1st_place_medal:

### CODE REVIEW

**Comment by MatthewJWalls** [comment]
PR build results: 
* https://s3-eu-west-1.amazonaws.com/prbuilds/PR-15514/screenshots/mobile.png 
* https://s3-eu-west-1.amazonaws.com/prbuilds/PR-15514/screenshots/wide.png 
* https://s3-eu-west-1.amazonaws.com/prbuilds/PR-15514/screenshots/tablet.png 
* https://s3-eu-west-1.amazonaws.com/prbuilds/PR-15514/screenshots/desktop.png 

 -automated message

File: common/app/views/main.scala.html [hunk 1]
```diff
@@ -7,18 +7,12 @@
...
     @if(!page.metadata.shouldHideHeaderAndTopAds) {
-        @if(showAdverts) {
-            @defining(
-                page match {
-                    case c: model.ContentPage => Some(c.item.tags)
-                    case _ => None
-                }
-            ) { maybeTags =>
-                @fragments.commercial.topBanner(page.metadata, edition, maybeTags)
+        <div class="bannerandheader">
```
Comment (rich-nguyen) [inline common/app/views/main.scala.html]: is `bannerandheader` supposed to be an id, or a class?

File: static/src/javascripts/bootstraps/commercial.js [hunk 2]
```diff
@@ -89,7 +89,8 @@ define([
             ['cm-hostedOJCarousel', hostedOJCarousel.init]);
     }
 
-    if (!config.page.disableStickyTopBanner) {
+    var supportsSticky = document.documentElement.classList.contains('has-sticky');
+    if (!config.page.disableStickyTopBanner && !supportsSticky) {
```
Comment (rich-nguyen) [inline static/src/javascripts/bootstraps/commercial.js]: Do you think its a little confusing now with have several sticky flags per page? I count three, `top-banner-ad-container--not-sticky` for the container, `!has-sticky` on the document, and `disableStickyTopBanner` in JS config.

**Comment by rich-nguyen** [comment]
👍  apart from comment

**Review by rich-nguyen — APPROVED** [review]

File: static/src/javascripts/bootstraps/commercial.js (hunk 2 above)
Comment (regiskuckaertz) [inline static/src/javascripts/bootstraps/commercial.js]: Yes I do 😲  I don't know how we'd do without them though, except if one day we draw a line in the sand and stop making the banner stick on those browsers that don't support `position: sticky`. But, to match the way other modules work, I could just add a flag in `commercialFeatures`. Let's do that!

File: common/app/views/main.scala.html (hunk 1 above)
Comment (regiskuckaertz) [inline common/app/views/main.scala.html]: That's an id, good call

**Review by kelvin-chappell — APPROVED** [review]
👍

**Comment by regiskuckaertz** [comment]
weird fact: even though Opera is advertised as supporting sticky, it is not so. In Opera 42:

picture 4

**Comment by regiskuckaertz** [comment]
Mandatory GIF, on Safari:

sticky-top

The stickiness is just so much more seamless.

**Pull request merged**